### PR TITLE
Fix system profile display issues

### LINKED ISSUE #3407: A bug with "system" profile
> Found this strange issue, but I think it's right to report it.
## Issue
Opening the _system_ profile blocks every profile opened after that.

## How to replicate this
1.  Send any command in chat in order to get the "system" message (using _/help_ for example):
help command
2. Open _system_ profile:
system profile
3. Close _system_ profile and open any other profile (_system_'s one too) to get the error and the _system_ profile again:
error

## Demo Video
https://streamable.com/6bfqr

## Logs
**runtime.log**
```
10/09/2018 17:29:27: Screen changed → Disclaimer
10/09/2018 17:29:33: Screen changed → Intro
10/09/2018 17:29:36: Screen changed → MainMenu
10/09/2018 17:29:53: ButtonSystem's state changed from Initial to TopLevel
10/09/2018 17:30:12: Focus contention triggered by NotificationOverlay (8,0) 320x768.
10/09/2018 17:30:19: Focus contention triggered by NotificationOverlay (172,0) 320x768.
10/09/2018 17:30:37: Focus contention triggered by UserProfileOverlay (0,0) 1.052x728.
10/09/2018 17:30:41: Unhandled exception has been allowed with 0 more allowable exceptions .
10/09/2018 17:30:43: An unhandled error has occurred.
10/09/2018 17:30:43: System.NullReferenceException: Object reference not set to an instance of an object.
10/09/2018 17:30:43: at osu.Game.Overlays.UserProfileOverlay.ShowUser(User user, Boolean fetchOnline)
10/09/2018 17:30:43: at osu.Framework.Graphics.Containers.ClickableContainer.OnClick(InputState state)
10/09/2018 17:30:43: at osu.Framework.Input.MouseButtonEventManager.<>c__DisplayClass33_0.<HandleMouseClick>b__0(Drawable t)
10/09/2018 17:30:43: at System.Linq.Enumerable.TryGetFirst[TSource](IEnumerable`1 source, Func`2 predicate, Boolean& found)
10/09/2018 17:30:43: at osu.Framework.Input.MouseButtonEventManager.HandleMouseClick(InputState state)
10/09/2018 17:30:43: at osu.Framework.Input.MouseButtonEventManager.HandleButtonStateChange(InputState state, ButtonStateChangeKind kind, Double currentTime)
10/09/2018 17:30:43: at osu.Framework.Input.InputManager.HandleMouseButtonStateChange(InputState state, MouseButton button, ButtonStateChangeKind kind)
10/09/2018 17:30:43: at osu.Framework.Input.StateChanges.ButtonInput`1.Apply(InputState state, IInputStateChangeHandler handler)
10/09/2018 17:30:43: at osu.Framework.Input.InputManager.Update()
10/09/2018 17:30:43: at osu.Framework.Input.PassThroughInputManager.Update()
10/09/2018 17:30:43: at osu.Framework.Graphics.Drawable.UpdateSubTree()
10/09/2018 17:30:43: at osu.Framework.Graphics.Containers.CompositeDrawable.UpdateSubTree()
10/09/2018 17:30:43: at osu.Framework.Platform.GameHost.UpdateFrame()
10/09/2018 17:30:43: at osu.Framework.Threading.GameThread.ProcessFrame()
10/09/2018 17:30:43: at osu.Framework.Threading.GameThread.runWork()
10/09/2018 17:30:43: Focus contention triggered by UserProfileOverlay (0,0) 1.052x728.
10/09/2018 17:30:44: Focus contention triggered by NotificationOverlay (0,0) 320x768.
```
(there's nothing interesting in the other logs)

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/63235809-a-bug-with-system-profile?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F4707269&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by UselessToucan**
Resolves #3407 
https://youtu.be/nydPyZf8W7M

### CODE REVIEW

**Comment by UselessToucan** [comment]
It is better not being able to open system profile at all, but as far as I understand there is no guaranteed way to distinct it from the real user. Something like `technical` boolean property.

File: osu.Game/Overlays/UserProfileOverlay.cs [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void ShowUser(User user, bool fetchOnline = true)
         {
             Show();
```
Comment (Aergwyn) [inline osu.Game/Overlays/UserProfileOverlay.cs]: What about moving this behind the check. Then it shouldn't open at all. Opening empty profiles feels really weird anyways.

**Comment by peppy** [comment]
There's "no way" to detect system. Except it has the default user ID which is never used by an actual use. So why not check against that?

**Comment by peppy** [comment]
Closing (merged #3419 instead).

**Pull request closed without merging**